app/src/main/java/org/oppia/android/app/recyclerview/BindableAdapter.kt

```diff
@@ -95,63 +96,15 @@ class BindableAdapter<T : Any> internal constructor(
     internal abstract fun bind(data: T)
   }
 
-  /**
-   * The base builder for [BindableAdapter]. This class should not be used directly--use either
-   * [SingleTypeBuilder] or [MultiTypeBuilder] instead.
-   */
-  abstract class BaseBuilder<BuilderType : Any> {
-    /**
-     * A [WeakReference] to a [LifecycleOwner] for databinding inflation. See [setLifecycleOwner].
-     * Note that this needs to be a weak reference so that long-held references to the adapter do
-     * not potentially leak lifecycle owners (such as fragments and activities).
-     */
-    private var lifecycleOwnerRef: WeakReference<LifecycleOwner>? = null
-
-    /**
-     * Sets the [LifecycleOwner] corresponding to this adapter. This will automatically be used as
-     * the lifecycle owner for all databinding classes created during view inflation. Note that the
-     * adapter holds a weak reference to the owner to ensure long-lived references to the adapter
-     * class itself does not result in leaks, however it's up to the caller's responsibility to make
-     * sure that the adapter itself is not actually used after the lifecycle owner has expired
-     * (otherwise the app may crash).
-     *
-     * @return this
-     */
-    fun setLifecycleOwner(lifecycleOwner: LifecycleOwner): BuilderType {
-      check(lifecycleOwnerRef == null) {
-        "A lifecycle owner has already been bound to this adapter."
-      }
-      lifecycleOwnerRef = WeakReference(lifecycleOwner)
-
-      // This cast is not, strictly speaking, safe, but child classes are expected to inherit from
-      // the builder & pass their own type in.
-      @Suppress("UNCHECKED_CAST") return this as BuilderType
-    }
-
-    /**
-     * Returns the [LifecycleOwner] bound to this adapter, or null if there isn't one. This method
-     * will throw if there was a lifecycle owner bound but is now expired.
-     */
-    protected fun getLifecycleOwner(): LifecycleOwner? {
-      // Crash if the lifecycle owner has been cleaned up since it's not valid to use the adapter
-      // with an old lifecycle owner, and silently ignoring this may result in part of the layout
-      // not responding to events.
-      return lifecycleOwnerRef?.let { ref ->
-        checkNotNull(ref.get()) {
-          "Attempted to inflate data binding with expired lifecycle owner"
-        }
-      }
-    }
-  }
-
   /**
    * Constructs a new [BindableAdapter] that for a single view type.
    *
-   * Instances of [SingleTypeBuilder] should be instantiated using [newBuilder].
+   * Instances of [SingleTypeBuilder] should be instantiated using [Factory].
    */
   class SingleTypeBuilder<T : Any>(
```

Comment:
Can we make this & the multi-type version a private constructor since it's only being called internally?

Reply:
I guess then we would not be able to access this inside inline create function, I am getting this error aftermaking it private constructor - 
```kotlin
Public-API inline function cannot access non-public-API 'private constructor SingleTypeBuilder<T : Any>(dataClassType: KClass<T#1 (type parameter of org.oppia.android.app.recyclerview.BindableAdapter.SingleTypeBuilder.Factory.create)>, lifecycleOwner: LifecycleOwner) defined in org.oppia.android.app.recyclerview.BindableAdapter.SingleTypeBuilder'
```